test/integration/authenticated-rest-api/authenticated-rest-api.test.ts

```diff
@@ -0,0 +1,136 @@
+import axios from "axios";
+import { APIGateway } from "aws-sdk";
+
+const api = new APIGateway();
+
+const TALIS_CDK_AUTH_API_VALID_CLIENT =
+  process.env.TALIS_CDK_AUTH_API_VALID_CLIENT ?? "";
+const TALIS_CDK_AUTH_API_VALID_SECRET =
+  process.env.TALIS_CDK_AUTH_API_VALID_SECRET ?? "";
```

Comment:
Do we want empty string literals here? What do they mean?

Reply:
When the env vars are not set, they are empty and the tests fail.

Happy to change this, but at some point we have to deal with the type of `process.env.TALIS_...` being `string | undefined`. Where these constants are being used, the type is `string`.

Is there a preferred way of reading env vars and using them as strings?

What is your suggestion here?

Reply:
I'm generally not a fan of coercing variables to silence type-system feedback. I'd typically view empty string literals as a code smell.

You could validate the environment first - save yourself even running the tests - a basic condition on the line below, or a validation function, perhaps. For something a little more comprehensive, you could do something like Grzeg's validated env in Content Linking:

https://github.com/techfromsage/content-linking-monorepo/blob/main/apps/api/src/config/config.ts

Or you could carry on with empty string literals - the test still fails. YMMV.